### PR TITLE
New attention mechanisms

### PR DESCRIPTION
**Pull request by Mr6one**

### CODE REVIEW

File: src/utils/__init__.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def parse_args():
     return params
 
 
-parse_args()
+# parse_args()
```
Comment (NonameUntitled) [inline src/utils/__init__.py]: Why did you do this? This call is needed for `GLOBAL_TENSORBOARD_WRITER` initialization. Revert this plz

```suggestion
parse_args()
```

File: src/utils/__init__.py [hunk 2]
```diff
@@ -1,5 +1,5 @@
-from utils.registry import MetaParent
-from utils.tensorboards import GLOBAL_TENSORBOARD_WRITER, TensorboardWriter
+from .registry import MetaParent
```
Comment (NonameUntitled) [inline src/utils/__init__.py]: Let's write relative paths only in \_\_init\_\_.py files. 

Quote from PEP8 guidelines: "Absolute imports are recommended, as they are usually more readable and tend to be better behaved (or at least give better error messages) if the import system is incorrectly configured (such as when a directory inside a package ends up on sys.path)". 

```suggestion
from utils.registry import MetaParent
```

File: src/utils/__init__.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
...
+from .tensorboards import GLOBAL_TENSORBOARD_WRITER, TensorboardWriter
```
Comment (NonameUntitled) [inline src/utils/__init__.py]: Same issue

```suggestion
from utils.tensorboards import GLOBAL_TENSORBOARD_WRITER, TensorboardWriter
```

File: src/models/baseline.py [hunk 4]
```diff
@@ -1,12 +1,51 @@
-from models.base import TorchModel
+from .base import TorchModel
```
Comment (NonameUntitled) [inline src/models/baseline.py]: Relative imports

```suggestion
from models.base import TorchModel
```

File: src/models/baseline.py [hunk 5]
```diff
@@ -1,12 +1,51 @@
...
 
-from utils import get_activation_function
-from utils import DEVICE
+from ..utils import get_activation_function
```
Comment (NonameUntitled) [inline src/models/baseline.py]: Relative imports

```suggestion
from utils import get_activation_function
```

File: src/models/baseline.py [hunk 6]
```diff
@@ -1,12 +1,51 @@
...
+from ..utils import DEVICE
```
Comment (NonameUntitled) [inline src/models/baseline.py]: Relative imports

```suggestion
from utils import DEVICE
```

File: src/models/baseline.py [hunk 7]
```diff
@@ -173,12 +212,12 @@ class BaselineEncoder(torch.nn.Module):
 
     def __init__(
             self,
+            attention,
```
Comment (NonameUntitled) [inline src/models/baseline.py:215]: What is attention? Can you explain please?

File: src/models/baseline.py [hunk 8]
```diff
@@ -1,12 +1,51 @@
...
-from utils import get_activation_function
-from utils import DEVICE
+from ..utils import get_activation_function
+from ..utils import DEVICE
 
 import torch
 import torch.nn as nn
 
 
+class TransformerEncoderLayer(nn.Module):
```
Comment (NonameUntitled) [inline src/models/baseline.py:10]: Why you decided to write implementation of `TransformerEncoderLayer` here? Can you explain please?

File: src/models/performer.py [hunk 9]
```diff
@@ -1,5 +1,35 @@
-from models.base import TorchModel
+from .base import TorchModel
```
Comment (NonameUntitled) [inline src/models/performer.py]: Relative imports

```suggestion
from models.base import TorchModel
```

File: src/models/informer.py [hunk 10]
```diff
@@ -1,4 +1,466 @@
-from models.base import TorchModel
+from .base import TorchModel
```
Comment (NonameUntitled) [inline src/models/informer.py]: Can you please revert `informer.py` since it was not your responsibility?

File: src/models/__init__.py [hunk 11]
```diff
@@ -1,2 +1,4 @@
 from .base import BaseModel
-from .baseline import BaselineModel
+from .baseline import BaselineModel, BaselineEncoder
+from .informer import InformerAttention
+from .performer import PerformerAttention, Performer
```
Comment (NonameUntitled) [inline src/models/__init__.py]: Empty line at EOF is missing

File: src/models/base.py [hunk 12]
```diff
@@ -1,4 +1,4 @@
-from utils import MetaParent
+from ..utils import MetaParent
```
Comment (NonameUntitled) [inline src/models/base.py]: Relative imports

```suggestion
from utils import MetaParent
```

**Review by NonameUntitled — CHANGES_REQUESTED** [review]

File: src/models/baseline.py (hunk 8 above)
Comment (Mr6one) [reply src/models/baseline.py:10]: Resolved in DM

**Pull request merged**